### PR TITLE
Create new Issue templates

### PR DESCRIPTION
**Pull request by cmgorton**
Adding a config.yml to give more options in our issues template. Will likely add some for documentation in the near future.

## What type of PR is this? (check all applicable)

- [X ] Issue template update

## Description
After discussing our issue templates in our forem/forem repo with Peter, we decided to add more templates to better direct people to the appropriate places for bug reports, features, and discussions. I will add a documentation link in the near future once we have the developer docs finalized.

## Added/updated tests?

- [ ] Yes
- [ x] No, and this is why: No tests needed to add issue config file to the repo. 
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [X ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

We're here to make it better

### CODE REVIEW

**Comment by cmgorton** [comment]
For more information on how I created this config file to include external links in our issue template you can see the documentation for it [here](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser)

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+blank_issues_enabled: false
+contact_links:
+  - name: Feature Request
+    url: https://forem.dev/t/features
+    about: Please use the `Create Post` feature request template to suggest an idea to make Forem better. 
+  - name: Forem Questions and Discussions
+    url: https://forem.dev/
+    about: Please ask and answer questions, discuss features, or reach out for help on forem.dev
+  - name: Self Host Bug Report
+    about: Create a report to help us improve the Self Host repository.
```
Comment (Ridhwana) [inline .github/ISSUE_TEMPLATE/config.yml:10]: **(nitpick, non-blocking)**: Can we swap the order of the `about` and `url` keys  in the `Self Host Bug Report` to keep it consistent with the order of the other points?

**Review by Ridhwana — APPROVED** [review]
Looks good to me 🚀, left one tiny nitpick but non-blocking suggestion 😊

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (cmgorton) [reply .github/ISSUE_TEMPLATE/config.yml]: Oops! I totally missed that. Yeah I think it should be swapped anyway. Thanks for catching that. I switched it.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+contact_links:
+  - name: Feature Request
+    url: https://forem.dev/t/features
+    about: Please use the `Create Post` feature request template to suggest an idea to make Forem better. 
+  - name: Forem Questions and Discussions
+    url: https://forem.dev/
+    about: Please ask and answer questions, discuss features, or reach out for help on forem.dev
+  - name: Self Host Bug Report
+    url: https://github.com/forem/selfhost/issues/new
+    about: Create a report to help us improve the Self Host repository.
```
Comment (mstruve) [inline .github/ISSUE_TEMPLATE/config.yml:11]: Should Self Host here and above have a dash, `Self-Host` like the [repo](https://github.com/forem/selfhost)?

**Review by mstruve — APPROVED** [review]
NICE!

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
Comment (cmgorton) [reply .github/ISSUE_TEMPLATE/config.yml:11]: Yeah I'll change that. 😄

**Pull request merged**